arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -1286,6 +1303,10 @@ auto ExecutionBlockImpl<Executor>::shadowRowForwarding(AqlCallStack& stack)
       // we need to forward them
       return ExecState::SHADOWROWS;
     } else {
+      if (hasDoneNothing and not shadowRow.isRelevant()) {
```

Comment:
Why is this handling not Needed in the other NEXTSUBQUERY case? (5 lines above, cannot comment there...)

Happy to discuss this offline, I have gut feeling that the above place is equally in danger.